bagel/utilities/derivative_utils.py

```diff
@@ -31,8 +31,8 @@ def get_recognized_pipelines(pipelines: Iterable[str]) -> list:
     )
 
     unrecognized_pipelines_details = (
-        f"Unrecognized processing pipelines: {unrecognized_pipelines}\n"
-        f"Supported pipelines are those in the Nipoppy pipeline catalog (https://github.com/nipoppy/pipeline-catalog):\n"
+        f"Unrecognized processing pipelines: {unrecognized_pipelines}. "
+        f"Supported pipelines are found in the Nipoppy pipeline catalog (https://github.com/nipoppy/pipeline-catalog): "
```

Comment:
are you stripping all these newlines on purpose?

Reply:
Yes - from what I've read, multi-line logs are generally discouraged since they can be hard to parse by log processors which usually are prepared to handle single-line logs. Multi-line logs only have the log prefix on the first line (unless we split them into different logs, but I don't think that makes much sense here since the messages are linked) and thus may require extra processing, and also results in unevenly aligned log messages 🥲 

That said, I think there may still be situations in which we need to log a multi-line message, such as full traceback, but I like the general principle of trying to keep logs to one line (esp when they require each other as context as above)!